spec/aptible/cli/subcommands/apps_spec.rb

```diff
@@ -197,6 +197,42 @@ def explain
     end
   end
 
+  describe '#apps:rename' do
+    before do
+      allow(Aptible::Api::App).to receive(:all) { [app] }
+      allow(Aptible::Api::Account).to receive(:all) { [account] }
+    end
+
+    before(:each) do
+      allow(subject).to receive(:options)
+        .and_return(environment: account.handle)
+    end
+
+    context 'with environment and app' do
+      it 'should rename properly' do
+        expect(app).to receive(:update)
+          .with(handle: 'hello2').and_return(app)
+        subject.send('apps:rename', 'hello', 'hello2')
+        expect(captured_logs).to include(
+          'In order for the new app name (hello2) to appear in log drain and '\
+          'metric drain destinations, you must restart the app.'
+        )
+      end
+      it 'should fail if app does not exist' do
+        expect { subject.send('apps:rename', 'hello2', 'hello3') }
+          .to raise_error(/Could not find app hello/)
+      end
+      it 'should raise error if update fails' do
+        response = Faraday::Response.new(status: 500)
+        error = HyperResource::ClientError.new('Mock Error', response: response)
+        expect(app).to receive(:update)
+          .with(handle: 'hello2').and_raise(error)
+        expect { subject.send('apps:rename', 'hello', 'hello2') }
+          .to raise_error(/Mock Error/)
+      end
```

Comment:
What error do we expect to be raised? I tried renaming two apps to the same name, but the second command didn't fail:

```
[aptible-cli]$ test-alex aptible apps
=== test-515
deploy-api
httpd-alpine

[aptible-cli]$ test-alex aptible apps:rename httpd-alpine foo
In order for the new app name (foo) to appear in log drain and metric drain destinations, you must restart the app.
You can restart your app with this command: "aptible restart --app foo --environment test-515"
Warning - Git remote addresses must be updated to match the new handle, if using Dockerfile deploy. (git@beta.aptible.com:madhu-sandbox/foo.git)

[aptible-cli]$ test-alex aptible apps:rename deploy-api foo
In order for the new app name (foo) to appear in log drain and metric drain destinations, you must restart the app.
You can restart your app with this command: "aptible restart --app foo --environment test-515"
Warning - Git remote addresses must be updated to match the new handle, if using Dockerfile deploy. (git@beta.aptible.com:madhu-sandbox/foo.git)

[aptible-cli]$ test-alex aptible apps
=== test-515
deploy-api
foo
```

My API returned 422:

```
rails-soothing-scarlet.test-alex  method=PUT path=/apps/3 format=hal controller=AppsController action=update status=422 error='ActiveRecord::RecordInvalid: Validation failed: Handle has already been taken, Handle has already been taken' duration=31.03 view=0.00 db=12.99 params={"handle"=>"foo"} params_truncated=false pid=231077 time=2022-08-30T00:20:36+00:00 remote_ip=[136.36.154.83](https://my.papertrailapp.com/groups/617523/events?q=%22136.36.154.83%22&focus=1501064283616907264&selected=1501064283616907264) token_id=ff8f4558-d279-481f-ad9e-a2b7c562ae04 user_agent="aptible-cli v0.19.3"
```

Reply:
any error to be bubbled up from the API calls.

i will try to use this as a mock response (422) and make sure the command fails

Reply:
With most recent changes:

When trying to update app with same name and causing failure (`apps:rename test-app-deploy madhu-test-cli --environment madhu-sandbox` when it already exists)

<img width="893" alt="image" src="https://user-images.githubusercontent.com/2961973/187515248-28f4624d-07bb-4a5a-88d5-d6a816acabff.png">

When trying to update db with same name and causing failure (`db:rename pg-2 pg-1 --environment madhu-sandbox` when `pg-1` already exists):

<img width="605" alt="image" src="https://user-images.githubusercontent.com/2961973/187515025-18ca6979-48f8-4491-b710-0511cfa555eb.png">

Reply:
Strangely the error bubbles up twice for apps 🤔 but NOT dbs

Reply:
This happens when I try to create an App with the same handle so this is an existing issue.

```
> aptible apps:create operation-logger-new
Validation failed: Handle has already been taken, Handle has already been taken
```